lib/azure/armrest/virtual_machine_service.rb

```diff
@@ -17,14 +17,17 @@ def initialize(configuration, options = {})
       end
 
       # Return a list of available VM series (aka sizes, flavors, etc), such
-      # as "Basic_A1", though information is included as well.
+      # as "Basic_A1", though other information is included as well.
       #
       def series(location)
-        unless @@providers_hash[provider.downcase] && @@providers_hash[provider.downcase]['locations/vmsizes']
-          raise ArgumentError, "Invalid provider '#{provider}'"
+        namespace = 'microsoft.compute'
+        resources = Azure::Armrest::Configuration.providers[namespace]
```

Comment:
As mentioned above you can send to get private `providers`
Service class already have method `resources`. Are they the same? If not, better rename `resources` variable here.